### PR TITLE
Add routing to abstract each component into it's own 'page'

### PR DESCRIPTION
**Pull request by barrymcgee**
## Done

Added [React Router](https://reacttraining.com/react-router/) so app can navigate between different "pages" to show different components

## Note

I have added a small bit of temporary CSS to create the sidebar layout, the aspiration is to implement the correct Vanilla docs layout but that is beyond the scope of this task.

## QA

- Pull code
- Run `npm run start`
- View http://localhost:3000/
- Click different patterns in the sidebar
- Run `npm run lint` - should be no errors
- Run `npm run test`-  should be no errors

Fixes: https://app.zenhub.com/workspace/o/ubuntudesign/vanilla-squad/issues/97

### CODE REVIEW

**Comment by anthonydillon** [comment]
@barrymcgee could you fix up the conflict before I review.

File: .eslintrc.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
     "react/prefer-stateless-function": 0,
     "react/forbid-prop-types": 0,
-    "function-paren-newline": 0
+    "function-paren-newline": 0,
+    "arrow-body-style": 2
```
Comment (anthonydillon) [inline .eslintrc.json]: Is `2` a valid value for this property? Seem to be `always`, `as-needed` and `never`: https://eslint.org/docs/rules/arrow-body-style

File: src/components/App/App.scss [hunk 2]
```diff
@@ -3,4 +3,21 @@
...
 @include vanilla-base;
 @include vf-p-grid;
-@include vf-p-grid-modifications;
\ No newline at end of file
+@include vf-p-grid-modifications;
+
+// Temp layout
+aside,
+main {
+  float: left;
```
Comment (anthonydillon) [inline src/components/App/App.scss]: Might be a nice place to start using CSS Grid?

File: src/components/Aside/Aside.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { BrowserRouter as Router, NavLink } from 'react-router-dom';
+
+class Aside extends React.Component {
+  render() {
+    return (
+      <Router>
+        <aside>
+          <ul className="p-list">
+            <li><NavLink exact to="/buttons" activeClassName="active">Buttons</NavLink></li>
```
Comment (anthonydillon) [inline src/components/Aside/Aside.js]: Add `p-list__item`

File: src/components/Aside/Aside.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+            <li><NavLink exact to="/cards" activeClassName="active">Cards</NavLink></li>
+            <li><NavLink exact to="/switches" activeClassName="active">Switches</NavLink></li>
+          </ul>
```
Comment (anthonydillon) [inline src/components/Aside/Aside.js]: Good start but might be worth passing an array in the future and iterating over it to create the navigation.

File: src/index.js [hunk 5]
```diff
@@ -1,8 +1,14 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { BrowserRouter as Router } from 'react-router-dom';
```
Comment (anthonydillon) [inline src/index.js]: Lint is failing because your defining Router without using it in this view.

**Comment by anthonydillon** [comment]
The views are not changing. The address bar gets updated but I remain on the homepage.

File: .eslintrc.json (hunk 1 above)
Comment (barrymcgee) [reply .eslintrc.json]: I thought that too but when running it with "never" as a string returns;

screen shot 2017-10-18 at 14 28 15

File: src/components/Aside/Aside.js (hunk 4 above)
Comment (barrymcgee) [reply src/components/Aside/Aside.js]: Yep, that's the plan once I figure out how to do it 😅

**Comment by barrymcgee** [comment]
@anthonydillon Actioned all points.

**Pull request merged**